### PR TITLE
Copy over analytics settings.

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #7923 

## Relevant technical choices

Deviations from IB:
* Couldn't implement old settings values fallback in `get` method in Settings, as at this point options have already synced with default to fill in the missing keys, and we arrive at the problem of determining what's older chicken or the egg - are the current settings updated values, or are the analytics settings correct values and options were updated with defaults (all 4 keys have different value types, it is not realistic to determine by them not having a value (having an empty value), as one is boolean, another array, etc). So I added the filter, which will run when options are retrieved but before missing keys are filled by default values to ensure settings array always returns proper data, and there is a way to check if they are missing.
* Removed `get` method as it is not needed any more
* Test is implemented in `Analytics_4Test` instead of Settings, due to change mentioned above

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/base.js [hunk 1]
```diff
@@ -32,8 +32,8 @@ const baseModuleStore = Modules.createModuleStore( 'analytics-4', {
 	ownedSettingsSlugs: [
 		// TODO: These can be uncommented when Analytics and Analytics 4
 		// modules are officially separated.
-		// 'accountID',
 		// 'adsConversionID',
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics-4/datastore/base.js:35]: This is not an owned setting, let's remove this remaining comment.

File: tests/phpunit/integration/Modules/Analytics_4/SettingsTest.php [hunk 2]
```diff
@@ -66,12 +65,16 @@ public function test_get_default() {
 		$this->assertEqualSetsWithIndex(
 			array(
 				// TODO: These can be uncommented when Analytics and Analytics 4 modules are officially separated.
-				// 'accountID'              => '',
 				// 'adsConversionID'        => '',
```
Comment (eugene-manuilov) [inline tests/phpunit/integration/Modules/Analytics_4/SettingsTest.php:68]: Same here, let's remove this remaining part of the comment.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. Mostly looks good to me. Left a few minor comments with some small changes. Please, let me know if you have questions.

**Comment by zutigrm** [comment]
@eugene-manuilov Thank you for the feedback. PR updated

File: tests/backstop/reference/google-site-kit_Components_TextField_0_document_0_small.png
Comment (eugene-manuilov) [inline tests/backstop/reference/google-site-kit_Components_TextField_0_document_0_small.png:1]: Looks like vrt images should be already updated in the develop branch. Try to merge the latest changes into your branch. I think these changes in images should disappear.

File: includes/Modules/Analytics_4/Settings.php [hunk 3]
```diff
@@ -176,32 +184,43 @@ function( $dimension ) {
...
-	 * @return array Updated module settings with the current user ID as the ownerID setting.
+	 * @param array $option Analytics 4 settings.
+	 * @return array Missing Analytics 4 settings array, or empty array if no setting is missing.
 	 */
-	protected function update_owner_id_in_settings( $settings ) {
-		// This is a temporary solution to sync owner IDs between Analytics and Analytics 4 modules.
-		// The owner ID setting of the Analytics module is the source of truth for the Analytics 4 module.
-		// This will change when Analytics is sunset and we merge both modules into one.
-		( new Analytics_Settings( $this->options ) )->merge( array( 'ownerID' => get_current_user_id() ) );
+	protected function retrieve_missing_analytics_4_settings( $option ) {
```
Comment (eugene-manuilov) [inline includes/Modules/Analytics_4/Settings.php:199]: Let's probably drop `analytics_4` from the function name: `retrieve_missing_settings`.

File: includes/Modules/Analytics_4/Settings.php [hunk 4]
```diff
@@ -39,29 +39,24 @@ public function register() {
...
+				if ( is_array( $option ) ) {
+					$missing_settings = $this->retrieve_missing_analytics_4_settings( $option );
+
+					if ( ! empty( $missing_settings ) ) {
+						return $option + $missing_settings;
+					}
+				}
+
+				return $option;
+			},
```
Comment (eugene-manuilov) [inline includes/Modules/Analytics_4/Settings.php:57]: I think we can move this logic into the `retrieve_missing_...` method and use `add_filter( 'option_' . self::OPTION, array( $this, 'retrieve_missing_...' ) )`.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. Mostly looks good to me. Just added a few minor comments with tiny changes. Please, take a look.

**Comment by zutigrm** [comment]
@eugene-manuilov Thank you for the suggestions, I updated PR

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
@zutigrm, I still see vrt images changed. Could you please revert changes in those images?

File: includes/Modules/Analytics_4/Settings.php [hunk 5]
```diff
@@ -176,32 +175,51 @@ function( $dimension ) {
...
+	 * @return array Missing Analytics 4 settings array, or empty array if no setting is missing.
 	 */
-	protected function update_owner_id_in_settings( $settings ) {
-		// This is a temporary solution to sync owner IDs between Analytics and Analytics 4 modules.
-		// The owner ID setting of the Analytics module is the source of truth for the Analytics 4 module.
-		// This will change when Analytics is sunset and we merge both modules into one.
-		( new Analytics_Settings( $this->options ) )->merge( array( 'ownerID' => get_current_user_id() ) );
+	protected function retrieve_missing_settings( $option ) {
+		if ( ! is_array( $option ) ) {
+			return;
```
Comment (eugene-manuilov) [inline includes/Modules/Analytics_4/Settings.php:192]:
```suggestion
			return $option;
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. The last one comment with a suggestion, that I'll apply by myself.

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**